pkg/scheduler/actions/common/allocate.go

```diff
@@ -27,6 +27,22 @@ func AllocateJob(ssn *framework.Session, stmt *framework.Statement, nodes []*nod
 		return false
 	}
 
+	partitions, err := ssn.PartitionFn(job, nodes)
+	if err != nil {
+		log.InfraLogger.Errorf("Failed to partition nodes for job <%s/%s> in session %v, err: %v", job.Namespace, job.Name, ssn.UID, err)
+		return false
+	}
+
+	for _, partition := range partitions {
+		if allocateJobOnPartition(ssn, stmt, partition, job, isPipelineOnly, tasksToAllocate) {
+			return true
+		}
+	}
+	return false
+}
+
+func allocateJobOnPartition(ssn *framework.Session, stmt *framework.Statement, nodes []*node_info.NodeInfo,
+	job *podgroup_info.PodGroupInfo, isPipelineOnly bool, tasksToAllocate []*pod_info.PodInfo) bool {
 	defer ssn.CleanAllocationAttemptCache(job)
```

Comment:
Notice the `defer ssn.CleanAllocationAttemptCache(job)` in here that would now happen on each partition try, multiple times for a single job, as opposed to before that it was happen once per job